HWs/HW1/BWT/BWT/Program.cs

```diff
@@ -0,0 +1,167 @@
+
+        var result = new StringBuilder();
+        int lineEndNumber = 0;
+        for (int i = 0; i < text.Length; ++i)
+        {
+            result.Append(text[(rotations[i] + text.Length - 1) % text.Length]);
+
+            if (rotations[i] == 0)
+            {
+                lineEndNumber = i;
+            }
+        }
+
+        return (result.ToString(), lineEndNumber);
+    }
+
+    public static string Decode(string text, int matrixIndex)
+    {
+        if (text.Length == 0)
+        {
+            throw new ArgumentException("Text cannot be empty.");
+        }
+
+        var characterFrequencies = new int[_alphabetSize];
+        for (int i = 0; i < text.Length; ++i)
+        {
+            ++characterFrequencies[text[i]];
+        }
+
+        int sum = 0;
+        for (int i = 0; i < _alphabetSize; ++i)
+        {
+            sum += characterFrequencies[i];
+            characterFrequencies[i] = sum - characterFrequencies[i];
+        }
+
+        var nextSymbolsIndexes = new int[text.Length];
+        for (int i = 0; i < text.Length; ++i)
+        {
+            nextSymbolsIndexes[characterFrequencies[text[i]]] = i;
+            ++characterFrequencies[text[i]];
+        }
+
+        var nextSymbol = nextSymbolsIndexes[matrixIndex];
+        var result = new StringBuilder();
+        for (int i = 0; i < text.Length; ++i)
+        {
+            result.Append(text[nextSymbol]);
+            nextSymbol = nextSymbolsIndexes[nextSymbol];
+        }
+
+        return result.ToString();
+    }
+
+    static void Main()
+    {
+        Console.WriteLine("Hi, I can encode and decode a text using the Burrows-Wheeler transform.");
+        Console.Write("Choose what you need [encode/decode]: ");
+
+        string? action = "";
+
+        while (action != "0")
+        {
+            action = Console.ReadLine();
+            if (action == null)
+            {
+                throw new ArgumentNullException("Text cannot be null.");
+            }
+            action = action.ToLower();
+
+            switch (action)
+            {
+                case "encode":
+                    Console.Write("Good! Enter your text: ");
+                    string? textForEncode = Console.ReadLine();
+                    while (textForEncode == null)
+                    {
+                        textForEncode = Console.ReadLine();
+                    }
+
+                    var resultEncode = Encode(textForEncode);
+                    Console.WriteLine($"Converted text: {resultEncode.Item1}\nLine end number: {resultEncode.Item2}");
+                    Console.Write("Don't want to see the reverse transformation? (Enter DECODE): ");
+                    break;
+
+                case "decode":
+                    Console.Write("Good! Enter your text: ");
+                    string? textForDecode = Console.ReadLine();
+                    while (textForDecode == null || textForDecode == "")
```

Comment:
```suggestion
                    while (string.IsNullOrEmpty(textForDecode))
```
Для этого есть стандартный метод)